components/Buttons/src/ColorThemer/MDCButtonColorThemer.m

```diff
@@ -33,6 +33,24 @@ + (void)applySemanticColorScheme:(nonnull id<MDCColorScheming>)colorScheme
   button.disabledAlpha = 1.f;
 }
 
++ (void)applySemanticColorScheme:(nonnull id<MDCColorScheming>)colorScheme
+                    toFlatButton:(nonnull MDCFlatButton *)flatButton {
+  [flatButton setBackgroundColor:[colorScheme.primaryColor colorWithAlphaComponent:0.f]
+                        forState:UIControlStateNormal];
+  [flatButton setBackgroundColor:[colorScheme.primaryColor colorWithAlphaComponent:0.f]
+                        forState:UIControlStateHighlighted];
+  [flatButton setBackgroundColor:[colorScheme.primaryColor colorWithAlphaComponent:0.f]
+                        forState:UIControlStateSelected];
```

Comment:
I prefer to rely on fallback behavior - I think most clients will expect that. UIKit tends to document that same expectation. It's often worded like, "at a minimum set the title for the `.normal` state."  If we are going to pick out specific states (like `.disabled`), there should be a difference between that state and `.normal`.

Reply:
I had the same approach in terms of the background color, but I wanted to follow the params doc guidelines and there they stated primaryColor + opacity of 0.

Reply:
Yeah, the reason I set all the states is because if a client changed the color of a state prior to the theming, the theming wont override that color to its rightful theming.

Reply:
The params doc unfortunately doesn't have any awareness of a "clear" color, so they specify colors in terms of color + opacity. This isn't a color that is intended to be mapped by the color system because the opacity is always going to be 0, so for our purposes I believe it's sufficient to use `clearColor`.

Reply:
sgtm, will update accordingly

Reply:
So should all of the states be added to the map instead of nilling-out and relying on the fallback?

Reply:
after speaking to both, have moved to a fallback to .normal approach with .normal and .disabled set in the themer.